### PR TITLE
Add extensive metrics in puppet-etcd module

### PR DESCRIPTION
**Pull request by Buffer0x7cd**
This PR adds metrics parameter in etcd configuration file

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -77,4 +77,6 @@
   $log_package_levels = undef
 
   $journald_forward_enable = false
+  #metrics
+  $metrics = 'extensive'
```
Comment (vkhromov) [inline manifests/params.pp]: As far as I understand, values from this file are used as the default values for the Puppet module.
But for `--metrics` the default value should be `basic`: https://etcd.io/docs/v3.5/op-guide/configuration/#--metrics
Thus, after merging this PR, all `etcd` clusters will run with the new setting at the same time, which could potentially break all of them at the same time.
I think it's better (and safer) to set `metrics` to `basic` here, so after switching to the updated module, `etcd` will run with the same settings as before.
You then can set `metrics` to `extensive` via hiera while gradually rolling out the new setting to our etcd clusters.

**Review by vkhromov — CHANGES_REQUESTED** [review]

File: manifests/params.pp (hunk 1 above)
Comment (Buffer0x7cd) [reply manifests/params.pp]: Thanks for pointing, It make sense to set the default to basic and gradually move clusters to extensive type

File: manifests/init.pp [hunk 2]
```diff
@@ -287,7 +287,9 @@
   # logging
   $debug                       = $etcd::params::debug,
   $log_package_levels          = $etcd::params::log_package_levels,
-  $journald_forward_enable     = $etcd::params::journald_forward_enable
+  $journald_forward_enable     = $etcd::params::journald_forward_enable,
+  # metrics
+  $metrics =                   = $etcd::params::metrics
```
Comment (vkhromov) [inline manifests/init.pp:292]: So, now this `$metrics` variable has `basic` as the default value and can be overridden by whoever instantiates `etcd` class or via the hiera.
But I don't see that it's used anywhere.
Don't you need to set that `ETCD_METRICS` environment variable (as you did in the original diff) to actually change behaviour of `etcd` according to the value of `$metrics` variable?

**Review by vkhromov — CHANGES_REQUESTED** [review]

File: manifests/init.pp (hunk 2 above)
Comment (Buffer0x7cd) [reply manifests/init.pp:292]: I am overwriting the metric parameter here https://github.yelpcorp.com/sysgit/puppet/pull/4558, Also the metric option is passed as cli parameter, so we dont need the environment variable

Comment (vkhromov) [reply manifests/init.pp:292]: :+1:

**Review by vkhromov — APPROVED** [review]
lgtm

**Pull request merged**